lib/WP_Auth0_LoginManager.php

```diff
@@ -543,16 +543,41 @@ protected function query_vars( $key ) {
   /**
    * Get and store state returned from Auth0
    *
-   * @return object|null
+   * @param bool $decoded - `true` to return decoded state
+   *
+   * @return string|object|null
    */
-  protected function get_state() {
+  protected function get_state( $decoded = FALSE ) {
 
     if ( empty( $this->state ) ) {
-      $state_val = urldecode( $this->query_vars( 'state' ) );
+      // Get and store base64 encoded state
+      $state_val = $this->query_vars( 'state' );
+      $state_val = urldecode( $state_val );
+      $this->state = $state_val;
+
+      // Decode and store
       $state_val = base64_decode( $state_val );
-      $this->state = json_decode( $state_val );
+      $this->state_decoded = json_decode( $state_val );
+    }
+
+    if ( $decoded ) {
+      return is_object( $this->state_decoded ) ? $this->state_decoded : null;
+    } else {
+      return $this->state;
     }
-    return is_object( $this->state ) ? $this->state : null;
+  }
+
+  /**
+   * Check the state send back from Auth0 with the one stored in the user's browser
+   *
+   * @return bool
+   */
+  protected function validate_state() {
+    $valid = isset( $_COOKIE[ WPA0_STATE_COOKIE_NAME ] )
```

Comment:
Are we going to force people to set the state then? Or better put.. is this library setting a new state on every request?

Reply:
Setting a new state when initializing Lock, yes